### DISCUSSION TITLE
Hydration error when clicking the app while loading

### DISCUSSION THREAD

**Discussion by mastoj** [vercel/next.js#43180] [Help]
I have an application/poc I'm working on but struggle with hydration. If I just sit back and wait for the application to load it works just fine, but if I click anywhere on the page before it has hydrated I get a hydration error:

![image](https://user-images.githubusercontent.com/501424/203011470-abc943dd-7c71-4599-a267-e77377ca1f4c.png)

First I thought it was related to one of my components, but it seems like it doesn't matter where I click I will get the error when I click the page.

Any idea what might be going on or how I debug this kind of issue?

I do use the new app folder for the application.

**UPDATE:** Small repro repo: https://github.com/mastoj/next-hydration-error

**Comment by vitali-duckma** [comment]
i also got this error in every next js i start, i separate header & footer and then include them inside _app.js

**Comment by icyJoseph** [comment]
Got a repo to look at? I guess the click is triggering some UI element to come up? Did this ever happen in pages?

**Reply by mastoj** [reply]
No public repo, but I'm working on a small reproduction. It doesn't have to be that I click on a interactive element, that was how I discovered it, but I can click on anything to reproduce it.

**Reply by icyJoseph** [reply]
I just saw another project where, under CPU slowdown, tabbing into the UI caused an error too... Hopefully yours helps us figure out what's cooking here...

**Reply by mastoj** [reply]
Here is a smaller reproduction, https://github.com/mastoj/next-hydration-error.

Note that if you load the page without interacting with it nothing happens. If you click like crazy anywhere on the page you'll see an hydration error from time to time. Doesn't happen every time so I suspect timing of clicks matters.

**Reply by icyJoseph** [reply]
Is the fragment necessary to trigger this? I think I might know what it is...

**Reply by mastoj** [reply]
removed the fragments, and the issue is still there.

**Reply by icyJoseph** [reply]
I see, I was also able to do the same with this issue report, https://github.com/vercel/next.js/issues/43159 and suspect these are one of the same...

**Reply by mastoj** [reply]
Do you know of a workaround I can use in the meantime?

**Reply by icyJoseph** [reply]
No idea, this is one the things with these beta features, I'd say, try to follow up on the #43159 issue or wait for 13.0.5.

**Reply by transitive-bullshit** [reply]
@mastoj if we're seeing the same issue, the workaround I found was to add some `<Suspense>` boundaries around several of my top-level components.

**Reply by transitive-bullshit** [reply]
See https://github.com/transitive-bullshit/next-movie/compare/main...feature/remove-suspense-test for an example (this is removing them in order to repro the bug, but if you add them, the bug seems to go away)

**Reply by icyJoseph** [reply]
Wow, 

```jsx
            <body>
                <Suspense fallback={null}>
                    <div className="min-h-scren w-full bg-secondary-neutral font-elkjop-regular relative">
                        <div className="min-h-screen max-w-screen-2xl mx-auto bg-white w-full">
                            <Header />
                            {/* <QuickMenu quickMenuLinks={quickMenuLinks} /> */}

                            {children}
                        </div>
                    </div>
                </Suspense>
            </body>
```

This seems to silence it...  I tried wrapping children only, but it didn't do the trick.

**Reply by mastoj** [reply]
Does adding Suspense impact server side rendering using the new app folder structure? Also, it seems like you needed to add the boundary around the whole "clickable" area.

**Reply by transitive-bullshit** [reply]
@mastoj it definitely seems to add an extra render during hydration if you test w/ the react profiler. So not a solution by any means, but still a viable workaround in the meantime nonetheless.

**Reply by mastoj** [reply]
ok, good to know. This is not mission critical by any means yet... so I might wait to add the workaround and see if a fix to the root cause will arrive.

**Reply by mastoj** [reply]
Adding suspense didn't really help for me in the larger application, not sure why though.

**Comment by icyJoseph** [comment]
The issue is no longer reproducible in later v13 versions, I reckon v14 and onward do not have it either. If someone identifies such an issue again, submit an updated, minimal repro repository. Thanks 🎉 !
